torchaudio/prototype/transducer.py

```diff
@@ -0,0 +1,147 @@
+import torch
+from torch.autograd import Function
+from torch.nn import Module
+from torchaudio._internal import (
+    module_utils as _mod_utils,
+)
+
+__all__ = ["rnnt_loss", "RNNTLoss"]
+
+
+class _RNNT(Function):
+    @staticmethod
+    def forward(ctx, acts, labels, act_lens, label_lens, blank, reduction):
+        """
+        acts: Tensor of (batch x seqLength x labelLength x outputDim) containing output from network
+        labels: 2 dimensional Tensor containing all the targets of the batch with zero padded
+        act_lens: Tensor of size (batch) containing size of each output sequence from the network
+        label_lens: Tensor of (batch) containing label length of each example
+        """
+
+        device = acts.device
+        certify_inputs(acts, labels, act_lens, label_lens)
+
+        acts = acts.to("cpu")
+        labels = labels.to("cpu")
+        act_lens = act_lens.to("cpu")
+        label_lens = label_lens.to("cpu")
+
+        loss_func = torch.ops.warprnnt_pytorch_warp_rnnt.rnnt
+
+        grads = torch.zeros_like(acts)
+        minibatch_size = acts.size(0)
+        costs = torch.zeros(minibatch_size, dtype=acts.dtype)
+
+        loss_func(acts, labels, act_lens, label_lens, costs, grads, blank, 0)
+
+        if reduction in ["sum", "mean"]:
+            costs = costs.sum().unsqueeze_(-1)
+            if reduction == "mean":
+                costs /= minibatch_size
+                grads /= minibatch_size
+
+        costs = costs.to(device)
+        ctx.grads = grads.to(device)
+
+        return costs
+
+    @staticmethod
+    def backward(ctx, grad_output):
+        grad_output = grad_output.view(-1, 1, 1, 1).to(ctx.grads)
+        return ctx.grads.mul_(grad_output), None, None, None, None, None
+
+
+@_mod_utils.requires_module('_warp_transducer')
+def rnnt_loss(acts, labels, act_lens, label_lens, blank=0, reduction="mean"):
+    """RNN Transducer Loss
+
+    Args:
```

Comment:
I think the documentation could be improved a bit. It could also be useful to reference the paper.